File: src/main/java/org/manapart/enderports/EnderPorts.java
package org.manapart.enderports;

import net.minecraft.item.Item;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.fml.common.Mod;

@Mod(EnderPorts.MODID)
@Mod.EventBusSubscriber(modid = EnderPorts.MODID)
public class EnderPorts {

    public static final String MODID = "enderports";
    public static final Teleporter teleporter = createBlock();
    public static final TeleporterItem teleporterItem = createItem(teleporter);
    public static Item enderportsIcon = createIcon();

    public EnderPorts() {
        MinecraftForge.EVENT_BUS.register(this);
    }

    private static Item createIcon() {
        Item icon = new Item(new Item.Properties());
        icon.setRegistryName(MODID + ":ep_icon");
        return icon;
    }

    private static Teleporter createBlock() {
        Teleporter teleporter = new Teleporter();
        teleporter.setRegistryName("teleporter");
        return teleporter;
    }

    private static TeleporterItem createItem(Teleporter block) {
        TeleporterItem teleporter = new TeleporterItem(block);
        teleporter.setRegistryName("teleporteritem");
        return teleporter;
    }


}

File: src/main/java/org/manapart/enderports/EnderPorts.kt
package org.manapart.enderports

import net.minecraft.world.item.CreativeModeTabs
import net.minecraftforge.event.BuildCreativeModeTabContentsEvent
import net.minecraftforge.fml.common.Mod
import org.manapart.enderports.ModBlocks.TELEPORTER_BLOCK
import org.manapart.enderports.ModItems.TELEPORT_ITEM
import thedarkcolour.kotlinforforge.forge.MOD_BUS


const val MODID = "enderports"

@Mod(MODID)
object EnderPorts {

    init {
        ModBlocks.REGISTRY.register(MOD_BUS)
        ModItems.REGISTRY.register(MOD_BUS)
        ModEntities.REGISTRY.register(MOD_BUS)
        MOD_BUS.addListener(::buildContents)
    }

    fun buildContents(event: BuildCreativeModeTabContentsEvent) {
        if (event.tabKey == CreativeModeTabs.REDSTONE_BLOCKS) {
            event.accept(TELEPORT_ITEM)
            event.accept(TELEPORTER_BLOCK)
        }
    }

}